src/api/include/migraphx/migraphx.h

```diff
@@ -77,6 +77,13 @@ migraphx_status migraphx_shape_create(migraphx_shape_t* shape,
                                       size_t* lengths,
                                       size_t lengths_size);
 
+migraphx_status migraphx_shape_create_with_strides(migraphx_shape_t* shape,
+                                                   migraphx_shape_datatype_t type,
+                                                   size_t* lengths,
+                                                   size_t lengths_size,
+                                                   size_t* strides,
+                                                   size_t strides_size);
+
```

Comment:
This function still cannot handle the case of a scalar shape for the function set_input_parameter_shape(name, dims), where dims is a empty vector